docs/coding-practices.md

```diff
@@ -0,0 +1,105 @@
+# Coding Practices
+
+Author : Cédric Gautier
+Date : 2024-01-01
+
+## SUMMARY:
+
+Have a standard on coding when we build our application.
+
+## CONTEXT:
+
+There are multiple layers when creating a software in which there's a team of multiple developers.
+Everyone has a coding style, their own way to write software but we would like to have some practices that we should respect to be able to have omni-present default practice.
+
+## SOLUTION:
+
+### Linting
+
+First of all, there is a linting standard that we'd like to respect. We'll have one for both parts : backend and frontend
+
+#### Back-End
+
+In the back-end we'll be using golang-ci-lint that has a respected standard. We'll be using the default linting that it proposes but this may be subject to be modified in the futur.
+
+- [errcheck](https://github.com/kisielk/errcheck) : Errcheck is a program for checking for unchecked errors in Go code. These unchecked errors can be critical bugs in some cases.
+- [gosimple ⚙️](https://github.com/dominikh/go-tools/tree/master/simple) : Linter for Go source code that specializes in simplifying code.
+- [govet ⚙️](https://pkg.go.dev/cmd/vet) : Vet examines Go source code and reports suspicious constructs, such as Printf calls whose arguments do not align with the format string.
+- [ineffassign](https://golangci-lint.run/usage/linters/#:~:text=v1.0.0-,ineffassign,-Detects%20when%20assignments) : Detects when assignments to existing variables are not used.
+- [staticcheck ⚙️](https://golangci-lint.run/usage/linters/#:~:text=v1.0.0-,staticcheck,-%E2%9A%99%EF%B8%8F) : It's a set of rules from staticcheck. It's not the same thing as the staticcheck binary. The author of staticcheck doesn't support or approve the use of staticcheck as a library inside golangci-lint.
```

Comment:
Did not understand can you provide more details please.

Reply:
golangci-lint uses staticcheck but not it's not the same because staticcheck doesn't support or approve the use of staticcheck as a library inside golangci-lint. 

What is does is : Using static analysis, it finds bugs and performance issues, offers simplifications, and enforces style rules.

 https://staticcheck.dev/docs/